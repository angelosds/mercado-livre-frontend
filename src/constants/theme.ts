export const ROOT_RELATIVE_SIZE = 16
