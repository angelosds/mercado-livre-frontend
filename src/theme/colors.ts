import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  primary: '#fe6000',
  secondary: '#3483fa',
  gray900: '#333333',
  gray700: '#666666',
  gray500: '#999999',
  gray300: '#eeeeee',
  white: '#ffffff',
};
