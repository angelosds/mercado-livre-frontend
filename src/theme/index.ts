import { DefaultTheme } from 'styled-components';

import { pixelToRem } from '../utils';

export const theme: DefaultTheme = {
  colors: {
    primary500: '#fee600',

    secondary100: '#c6dcf7',
    secondary500: '#3483fa',
    secondary700: '#2968c8',

    gray900: '#333333',
    gray700: '#333333',
    gray500: '#666666',
    gray400: '#999999',
    gray300: '#bbbbbb',
    gray200: '#dddddd',
    gray100: '#eeeeee',

    border200: '#f7f7f7',

    success: '#1ed760',

    white: '#ffffff',
  },
  fontSizes: {
    xs: pixelToRem(14),
    sm: pixelToRem(16),
    md: pixelToRem(24),
    lg: pixelToRem(28),
    xl: pixelToRem(46),
  },
  outline: {
    sm: '3px'
  },
  space: {
    sm: '4px',
    md: '8px',
    lg: '16px',
    xl: '32px',
  },
  radius: {
    sm: '4px'
  },
  breakpoints: {
    mobile: '565px'
  }
};
