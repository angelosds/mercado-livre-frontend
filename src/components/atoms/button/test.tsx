import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components';

import { theme } from '../../../theme';

import Button, { Props } from '.'

describe('Button', () => {
  const build = ({ children, onClick }: Props) => {
    const { container } = render(<ThemeProvider theme={theme}><Button onClick={onClick}>{children}</Button></ThemeProvider>)

    return { container }
  }

  it('should render the component', () => {
    const { container } = build({ children: '', onClick: () => {} })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render button text', () => {
    const children = 'hello world'

    build({ children, onClick: () => {} })

    expect(screen.getByRole('button').textContent).toEqual(children)
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('should call onClick prop when user clicks the button', () => {
    const onClick = jest.fn()

    build({ children: '', onClick })

    screen.getByRole('button').click()

    expect(onClick).toHaveBeenCalled()
  })
})
