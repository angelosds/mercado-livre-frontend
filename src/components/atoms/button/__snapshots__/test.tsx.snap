// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button text 1`] = `
<button
  class="sc-bdvvtL gNXXAY"
  type="button"
>
  hello world
</button>
`;

exports[`Button should render the component 1`] = `
<button
  class="sc-bdvvtL gNXXAY"
  type="button"
/>
`;
